### DISCUSSION TITLE
Deploying Quarto on Google App Engine with Identity-Aware proxy

### DISCUSSION THREAD

**Discussion by dkapitan** [quarto-dev/quarto-cli#9831] [Show and Tell]
### Description

So I use Quarto a lot, and have been scratching my head for a while what the easiest and most robust way is to deploy it with simple user management.

I ended up with Google App Engine (GCP is my default cloud provider) with an Identiy-Aware proxy. [This blog post](https://cloud.google.com/blog/topics/developers-practitioners/control-access-your-web-sites-identity-aware-proxy) explains the basics of that solution design.

[This repository](https://github.com/dkapitan/quarto-with-iap) shows a minimal example, which is essentially just one extra `app.yaml` file that configures the app.

Very happy that this works, I spend quite some evenings figuring out a solution. I hope this is helpful for others, too.

**Comment by cscheid** [comment]
Thanks for sharing!

**Comment by mcanouil** [comment]
Next step is to setup a workflow in Cloud Build to build your website and deploy it to the App Engine. 😉
You can use the Docker image from the Quarto CLI repository to render and one of the Google ones with gcloud for the deployment.
(Can't share the setup, but it's not that hard to implement and works great, at least it has been for at least a year now)

**Reply by dkapitan** [reply]
Indeed!
